eth/filters/api.go

```diff
@@ -239,13 +239,13 @@ func (args *NewFilterArgs) UnmarshalJSON(data []byte) error {
 		return err
 	}
 
-	if raw.From == nil {
+	if raw.From == nil || raw.From.Int64() < 0 {
```

Comment:
The API supports passing in valid block numbers + latest + pending. If you pass in latest (-1), that is handled by the filter correctly. If you pass in pending (-2) that's not handled in the filter, which will convert it to uint64 (huge number) and will start iterating the blockchain util that huge number.